cannon/mipsevm/multithreaded/mips.go

```diff
@@ -0,0 +1,318 @@
+package multithreaded
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm"
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/exec"
+)
+
+func (m *InstrumentedState) handleSyscall() error {
+	thread := m.state.getCurrentThread()
+
+	syscallNum, a0, a1, a2, a3 := exec.GetSyscallArgs(m.state.GetRegisters())
+	v0 := uint32(0)
+	v1 := uint32(0)
+
+	//fmt.Printf("syscall: %d\n", syscallNum)
+	switch syscallNum {
+	case exec.SysMmap:
+		var newHeap uint32
+		v0, v1, newHeap = exec.HandleSysMmap(a0, a1, m.state.Heap)
+		m.state.Heap = newHeap
+	case exec.SysBrk:
+		v0 = exec.BrkStart
+	case exec.SysClone: // clone
+		// a0 = flag bitmask, a1 = stack pointer
+		if (a0 & exec.ValidCloneFlagsBitmask) != a0 {
+			panic(fmt.Sprintf("Unrecognized clone flags %b (supported flags: %b)", a0, exec.ValidCloneFlagsBitmask))
+		}
+
+		v0 = m.state.NextThreadId
+		v1 = 0
+		newThread := &ThreadState{
+			ThreadId:         m.state.NextThreadId,
+			ExitCode:         0,
+			Exited:           false,
+			FutexAddr:        exec.FutexEmptyAddr,
+			FutexVal:         0,
+			FutexTimeoutStep: 0,
+			Cpu: mipsevm.CpuScalars{
+				PC:     thread.Cpu.NextPC,
+				NextPC: thread.Cpu.NextPC + 4,
+				HI:     thread.Cpu.HI,
+				LO:     thread.Cpu.LO,
+			},
+		}
+		for i := 0; i < 32; i++ {
+			newThread.Registers[i] = thread.Registers[i]
+		}
```

Comment:
Arrays (`[32]uint32`) are value types in Go. So you can copy them like this:
```suggestion
		newThread.Registers = thread.Registers
```